File: ts/1408/1351.ts
// Z 形查找
function countNegatives(grid: number[][]): number {
  const m = grid.length, n = grid[0].length
  let cnt = 0, i = 0, j = n - 1
  while (i < m && j >= 0) {
    if (grid[i][j] < 0) {
      cnt += m - i
      j -= 1
    } else i += 1
  }
  return cnt
};

File: py3/1408/1351.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        n = len(grid[0])
        cnt = 0
        for row in grid:
            l, r = 0, n - 1
            while l <= r:
                m = l + (r - l) // 2
                if row[m] >= 0:
                    l = m + 1
                else:
                    r = m - 1
            cnt += n - l

        return cnt